### PR TITLE
ASP.NET Core (and other) cross-link updates

### LINKED ISSUE #3004: Request to update cross-links to ASP.NET Core docs
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

N/A

### Web app

Not Applicable

### Web API

Not Applicable

### Token cache serialization

Not Applicable

### Description

Hello ... I noticed in some API comments that some of the links to the ASP.NET Core docs are pinning to specific versions of our articles. If there's a specific reason to pin to a given version (e.g., a cross-linked section only exists or on applies by version), it's fine. In most cases, however, we don't want to pin cross-links to stale, prior-release content. Here's a link that shows where these instances live ...

https://github.com/search?q=repo%3AAzureAD%2Fmicrosoft-identity-web+%22%3Fview%3Daspnetcore-%22&type=code

All (or most) of these just need their query strings (e.g., `?view=aspnetcore-3.1`) stripped off.

Also, the culture segments (e.g., `en-us`) should be dropped. There are many more of these ...

https://github.com/search?q=repo%3AAzureAD%2Fmicrosoft-identity-web+%22en-us%22&type=code

... and the correct address now is "learn.microsoft.com." There are also many more of these to update ...

https://github.com/search?q=repo%3AAzureAD%2Fmicrosoft-identity-web+%22docs.microsoft.com%22&type=code

Example:

```diff
- https://docs.microsoft.com/en-us/aspnet/core/security/samesite?view=aspnetcore-3.1
+ https://learn.microsoft.com/aspnet/core/security/samesite
```

After that, the article version floats, and the article will load for the latest version. The culture floats, so the reader's preferred culture loads. The correct address prevents an unnecessary redirect.

Again, this request on pinning to a specific version only applies to when the API remarks don't pertain to a specific version of the article. Each API remark with a versioned cross-link has to be checked. I don't recommend updating them with a find-'n-replace gesture.

### Reproduction steps

N/A

### Error message

N/A

### Id Web logs

N/A

### Relevant code snippets

```csharp
N/A
```

### Regression

N/A

### Expected behavior

N/A

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #3004

The primary goal is to remove the version pinning to ASP.NET Core docs. I also ...

* Changed the site domain (`docs.microsoft.com` :point_right: `learn.microsoft.com`).
* Dropped the culture segments (i.e., `en-us`).

I didn't test all of the links, and I only performed a little work with text around links and link text.

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
This is great, thanks @guardrex

File: tests/DevApps/B2CWebAppCallsWebApi/Client/Startup.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public void ConfigureServices(IServiceCollection services)
                 // This lambda determines whether user consent for non-essential cookies is needed for a given request.
                 options.CheckConsentNeeded = context => true;
                 options.MinimumSameSitePolicy = SameSiteMode.Unspecified;
-                // Handling SameSite cookie according to https://docs.microsoft.com/en-us/aspnet/core/security/samesite?view=aspnetcore-3.1
+                // Handling SameSite cookie according to https://learn.microsoft.com/aspnet/core/security/samesite
```
Comment (msbw2) [inline tests/DevApps/B2CWebAppCallsWebApi/Client/Startup.cs:37]:
```suggestion
                // Handles SameSite cookies according to the ASP.NET Core documentation at https://learn.microsoft.com/aspnet/core/security/samesite.
```

Comment (guardrex) [reply tests/DevApps/B2CWebAppCallsWebApi/Client/Startup.cs:37]: There are several of these, so I'll do a find-'n-replace on my branch to get this phrasing change all in one go.

File: tests/DevApps/WebAppCallsWebApiCallsGraph/Client/Startup.cs [hunk 2]
```diff
@@ -37,7 +37,7 @@ public void ConfigureServices(IServiceCollection services)
                 // This lambda determines whether user consent for non-essential cookies is needed for a given request.
                 options.CheckConsentNeeded = context => true;
                 options.MinimumSameSitePolicy = SameSiteMode.Unspecified;
-                // Handling SameSite cookie according to https://docs.microsoft.com/en-us/aspnet/core/security/samesite?view=aspnetcore-3.1
+                // Handling SameSite cookie according to https://learn.microsoft.com/aspnet/core/security/samesite
```
Comment (kellyyangsong) [inline tests/DevApps/WebAppCallsWebApiCallsGraph/Client/Startup.cs:40]: Nit 
```suggestion
                // Handles SameSite cookie according to https://learn.microsoft.com/aspnet/core/security/samesite
```

File: tests/DevApps/B2CWebAppCallsWebApi/Client/Startup.cs (hunk 1 above)
Comment (guardrex) [reply tests/DevApps/B2CWebAppCallsWebApi/Client/Startup.cs:37]: ... and I won't forget the period on this one.

**Review by kellyyangsong — APPROVED** [review]
A couple small nits. Checked many of the new links and they look good. Thanks!

File: tests/DevApps/WebAppCallsWebApiCallsGraph/Client/Startup.cs (hunk 2 above)
Comment (msbw2) [inline tests/DevApps/WebAppCallsWebApiCallsGraph/Client/Startup.cs:40]:
```suggestion
                // Handles SameSite cookies according to the ASP.NET Core documentation at https://learn.microsoft.com/aspnet/core/security/samesite.
```

**Review by msbw2 — APPROVED** [review]

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @guardrex 
Very nice!

**Pull request merged**